asmcomp/cmm_helpers.ml

```diff
@@ -580,15 +584,25 @@ let rec remove_unit = function
       Ccatch(rec_flag, List.map map_h handlers, remove_unit body)
   | Ctrywith(body, exn, handler, dbg) ->
       Ctrywith(remove_unit body, exn, remove_unit handler, dbg)
-  | Clet(id, c1, c2) ->
-      Clet(id, c1, remove_unit c2)
+  | Clet(kind, id, def, body) ->
+      Clet(kind, id, def, remove_unit body)
+  | Cphantom_let(id, def, body) ->
+      Cphantom_let(id, def, remove_unit body)
   | Cop(Capply _mty, args, dbg) ->
       Cop(Capply typ_void, args, dbg)
   | Cop(Cextcall(proc, _mty, alloc, label_after), args, dbg) ->
       Cop(Cextcall(proc, typ_void, alloc, label_after), args, dbg)
   | Cexit (_,_) as c -> c
   | Ctuple [] as c -> c
-  | c -> Csequence(c, Ctuple [])
+  | (Cconst_int _ | Cconst_natint _
+    | Cconst_float _
+    | Cconst_symbol _
+    | Cconst_pointer _ | Cconst_natpointer _
+    | Cblockheader _
+    | Cvar _
+    | Cassign _
+    | Ctuple (_::_)
+    | Cop (_, _, _)) as c -> Csequence(c, Ctuple [])
```

Comment:
Actually, I wonder if the cases above `Cassign` need to build a sequence at all. The constants are pure and can be removed, and for `Cvar` the variable should have type `unit`, so removing it should not be observable at all.
One could also consider these cases to be bugs, but I haven't checked whether they can occur in practice.

The proposed code is fine for a refactoring PR, though.